app/src/main/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModel.kt

```diff
@@ -0,0 +1,155 @@
+package com.jalasoft.routesapp.ui.auth.login.viewModel
+
+import android.annotation.SuppressLint
+import android.app.AlertDialog
+import android.content.Context
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.google.firebase.auth.AuthCredential
+import com.google.firebase.auth.FirebaseAuth
+import com.jalasoft.routesapp.R
+import com.jalasoft.routesapp.data.model.remote.User
+import com.jalasoft.routesapp.data.remote.interfaces.IUserManager
+import com.jalasoft.routesapp.data.remote.managers.AuthFirebaseManager
+import com.jalasoft.routesapp.data.remote.managers.UserManager
+import com.jalasoft.routesapp.util.helpers.UserTypeLogin
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel
+@Inject
+constructor(private val userManager: UserManager, private val auth: AuthFirebaseManager) : ViewModel(), IUserManager {
+
+    val loginIsSuccessful: MutableLiveData<Boolean> by lazy {
+        MutableLiveData<Boolean>()
+    }
+    val errorMessage: MutableLiveData<String> by lazy {
+        MutableLiveData<String>()
+    }
+    val signInGoogleOrFacebook: MutableLiveData<Boolean> by lazy {
+        MutableLiveData<Boolean>()
+    }
+
+    @SuppressLint("StaticFieldLeak")
+    var context: Context? = null

```

Comment:
context should not be stored as a variable in the viewmodel, as the context can change from time to time.

Is better to have a context as a parameter in every place you need it, that way you don't have to silence the staticfieldleak

Reply:
This is from Adriana's code